index.bs

```diff
@@ -4629,14 +4886,33 @@ partial interface MLGraphBuilder {
         - *SumSquare*: Compute the sum of the square of all the input values along the axes.
 </div>
 
+<details open algorithm>
+  <summary>
+    To <dfn for="MLGraphBuilder">calculate reduction output sizes</dfn>, given a [=/list=] of unsigned integers |inputShape|, a optional [=/list=] of unsigned integers |axes|, and [=/boolean=] |keepDimensions|, perform the following steps. They return a new [=/list=] of unsigned integers, or failure.
+  </summary>
+  <div class=algorithm-steps>
+    1. Let |inputSize| be |inputShape|'s [=list/size=].
+    1. If |axes| is given, then if any its [=list/items=] are not in [=the range=] 0 to |inputSize|, exclusive, return failure.
```

Comment:
According to this [proposal](https://github.com/webmachinelearning/webnn/issues/572#issuecomment-1962198054), should the |axes| be validated in "Validate arguments:" section? Then, this method could assume the axes is present and valid.

Reply:
Ah, good point - I pulled it in to dedupe, but you're right - we need to be consistent. I pulled the failure mode back out.